cs_api_io/cs_api_io.ml

```diff
@@ -77,3 +77,25 @@ let get_response {Response.code; body} =
   else
     let message = Printf.sprintf "HTTP response: %d\n%s" code body in
     Lwt_result.fail message
+
+let get_error_message (errors : Yojson.Safe.t list) =
+  match CCList.hd errors with
+  | `Assoc json -> (
+    CCList.find_opt (fun (m, _) -> m = "message") json |> function
+    | Some (_, `String error_message) -> error_message
+    | _ -> "Unexpected response from the server")
+  | _ -> "Unexpected response from the server"
+
+let get_response_parse_error ({body; _} : Response.t) =
+  let open Lwt_result.Infix in
+  let json = Yojson.Safe.from_string body in
+  (match json with
+  | `Assoc res -> Lwt_result.return res
+  | _ -> Lwt_result.fail "Unexpected response from the server")
+  >|= CCList.find_opt (fun (m, _) -> m = "errors")
+  >>= function
+  | None -> Lwt_result.return body
+  | Some (_, errors_json) -> (
+    match errors_json with
+    | `List error_list -> get_error_message error_list |> Lwt_result.fail
+    | _ -> Lwt_result.fail "Unexpected response from the server")
```

Comment:
These two functions (`get_response_parse_error` and `get_error_message`) have some code that parses a JSON response and returns a error message. What about isolating that code in one OCaml function (e.g. `get_graphql_errors : Yojson.Safe.t -> string list`) and writing some tests for that function? That way it should be easier for me and other developers to be confident the code is correct.

By the way, can `get_response_parse_error` replace `get_response`? I don't understand why we still need the old one.

Reply:
I had some problem with some exceptions when I replaced get_response with get_response_parse_error in some cases. Also, get_response_parse_error was not made to handle the cases where the error came from the error code. Now it should work. Do you like it better this way :thinking: ?